tests/tests/tier0/proxy-service-start/test_proxy_service_start.py

```diff
@@ -1,47 +1,49 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 
 import re
+import os
 import pytest
 from typing import Dict
 
 from hirte_test.config import HirteControllerConfig, HirteNodeConfig
-from hirte_test.container import HirteContainer
+from hirte_test.container import HirteControllerContainer, HirteNodeContainer
 from hirte_test.test import HirteTest
-from hirte_test.util import read_file
 
 
 node_foo_name = "node-foo"
 node_bar_name = "node-bar"
 
 
-def exec(ctrl: HirteContainer, nodes: Dict[str, HirteContainer]):
+def verify_proxy_start(ctrl: HirteControllerContainer, nodes: Dict[str, HirteNodeContainer]):
     foo = nodes[node_foo_name]
     bar = nodes[node_bar_name]
 
-    foo.create_file("/etc/systemd/system", "requesting.service", read_file("./systemd/requesting.service"))
-    foo.systemctl_daemon_reload()
-    bar.create_file("/etc/systemd/system", "simple.service", read_file("./systemd/simple.service"))
-    bar.systemctl_daemon_reload()
+    source_dir = os.path.join(".", "systemd")
+    target_dir = os.path.join("/", "etc", "systemd", "system")
+    requesting_service = "requesting.service"
+    simple_service = "simple.service"
 
-    foo.wait_for_hirte_agent()
-    bar.wait_for_hirte_agent()
-    ctrl.wait_for_hirte()
+    foo.copy_systemd_service(requesting_service, source_dir, target_dir)
+    bar.copy_systemd_service(simple_service, source_dir, target_dir)
 
-    result, _ = ctrl.exec_run(f"hirtectl start {node_foo_name} requesting.service")
-    assert result == 0
+    result, output = ctrl.exec_run(f"hirtectl start {node_foo_name} {requesting_service}")
```

Comment:
I think you should enable the service first 
systemctl enable inside the container before start 

I think you should enable it after the copy

Reply:
This shouldn't be necessary. `systemctl enable` will ensure the unit is started on the next boot.